projects/@shared/lib/validate-data.js

```diff
@@ -0,0 +1,34 @@
+// Docs: https://joi.dev/api/?v=17.4.0#anyvalidatevalue-options
+const JOI_VALIDATION_OPTIONS = {
+  abortEarly: false,
+  convert: false,
+};
+
+const validateData = (data, schema) => {
+  if (!data.length) { return; }
```

Comment:
Could you once more think about a consistent return type for this function? Currently, it could be:
- `undefined`
- `{ errors: null, warnings: null }`
- `{ errors: /*actual errors*/, warnings: /*actual warning*/ }`
- whatever the error object is that is thrown by `Joi.object({/*...*/}).validate`